### ISSUE TITLE
MultipleSelect with ForeignKey

### ISSUE THREAD

**Issue by mdf-github** [digi604/django-smart-selects#178]
More of a question than an issue.

I have the following model:

```
class Country(models.Model):
    name = models.CharField(max_length=70)

class City(models.Model):
    name = models.CharField(max_length=70)
    country_code = models.CharField(max_length=3, primary_key=True, unique=True)
    country = ForeignKey(Country)
```

Since there's only one country a city belongs to, the relationship has to be ForeignKey. Further, I will be populating this list from a JSON file where the city is related to the country by `country_code`, so a ManyToMany relationship is not ideal.

What I want is for the user to select one country, and for that country, select one _**or more**_ cities at the same time. (For example, I want to select Tokyo and Osaka in Japan, but I don't want to see Beijing in the choices. Also, I don't want to click submit, get another blank form, and input Japan again to select the second city.) Normally, I would do this via a ManyToMany relationship, but given the situation as described above, I do not think it is ideal. If I have a Location model like:

```
class Location(models.Model):
    country = ForeignKey(Country)
    city = ChainedForeignKey(City, chained_field='country', chained_model_field='country')
```

how do I implement a multiple select form, which, upon submit, will create multiple instances of Location for each city selected? Is this feature supported in smart_selects, or is there a possible workaround?

Thanks a lot!

**Comment by blag** [comment]
If I understand your question correctly, this is where Django formsets would work for you. And yes, this should work for you.

**Comment by mdf-github** [comment]
Hi @blag, thanks for the suggestion! However, I tried formsets, and what I get is this:

Country 1: <select>
City 1: <select>
Country 2: <select>
City 2: <select>

What I would rather have is
Country: <select ONE only>
City: <select AT LEAST ONE, from the cities under the selected country>

so that at least I don't have to keep on selecting the same country for several cities. Here's my code for the forms:

```
class LocationForm(forms.ModelForm):
    class Meta:
         model = models.Location
         exclude = ('user',) ## I forgot to mention that the location has a ForeignKey to User as well
LocationFormSet = formset_factory(LocationForm, extra=2)
```

Am I doing it wrong? Sorry if this is a noob question -- I'm fairly new to Django.

**Comment by blag** [comment]
No, that is the expected result - you are doing it correctly. This would require some major changes, and I don't have too much time to work on that right now. If you get this working, please report back. I'll review any PR adding this functionality.

**Comment by blag** [comment]
This should be fixed in version 1.4.0. I'll close it but if it's not fixed please reopen it.
